File: backend/app/schemas/config_sections/game_core_config.py
from pydantic import BaseModel, Field, validator
import re


class GameCoreConfig(BaseModel):
    """Core game configuration and feature flags"""
    Version: str = Field(..., min_length=1, description="Game version (e.g., 1.0.0)")
    BuildNumber: int = Field(ge=1, description="Build number")
    MinSupportedVersion: str = Field(..., description="Minimum supported client version")
    ForceUpdate: bool = Field(default=False, description="Force client update")
    MaintenanceMode: bool = Field(default=False, description="Enable maintenance mode")
    MaintenanceMessage: str = Field(default="", description="Maintenance message to display")
    
    @validator('Version', 'MinSupportedVersion')
    def validate_version_format(cls, v: str) -> str:
        """Validate semantic version format (e.g., 1.0.0)"""
        pattern = r'^\d+\.\d+\.\d+$'
        if not re.match(pattern, v):
            raise ValueError('Version must be in format X.Y.Z (e.g., 1.0.0)')
        return v
    
    class Config:
        json_schema_extra = {
            "example": {
                "Version": "1.0.0",
                "BuildNumber": 100,
                "MinSupportedVersion": "1.0.0",
                "ForceUpdate": False,
                "MaintenanceMode": False,
                "MaintenanceMessage": ""
            }
        }

File: frontend/src/lib/validations/gameCoreConfig.ts
import { z } from 'zod';

const versionRegex = /^\d+\.\d+\.\d+$/;

export const gameCoreConfigSchema = z.object({
  version: z
    .string()
    .min(1, 'Version is required')
    .regex(versionRegex, 'Version must be in format X.Y.Z (e.g., 1.0.0)'),
  build_number: z.number().int().min(1, 'Build number must be at least 1'),
  min_supported_version: z
    .string()
    .min(1, 'Min supported version is required')
    .regex(versionRegex, 'Min supported version must be in format X.Y.Z (e.g., 1.0.0)'),
  force_update: z.boolean(),
  maintenance_mode: z.boolean(),
  maintenance_message: z.string(),
});

export type GameCoreConfig = z.infer<typeof gameCoreConfigSchema>;

